### PR TITLE
Adjust the font color in the status bar when Theia is offline

### LINKED ISSUE #2877: [status-bar][styling] Adjust the font color in the status bar when Theia is offline
Currently, it is the yellowish background with the white font. We should change it and use the [`--theia-warn-font-color0`](https://github.com/theia-ide/theia/blob/f93d2811e63555725ee999c7db1840d45352a434/packages/core/src/browser/style/variables-dark.useable.css#L127) so that we align the styles with Git and SIW.

screen shot 2018-09-13 at 17 48 39

### PR DESCRIPTION
**Pull request by lexcorp16**
Signed-off-by: Fasoro-Joseph Alexander <afasorojoseph@gmail.com>

Fixes #2877

### CODE REVIEW

File: packages/core/src/browser/connection-status-service.ts [hunk 1]
```diff
@@ -205,6 +205,7 @@ export class ApplicationConnectionStatusContribution extends DefaultFrontendAppl
 
     protected handleOffline() {
         this.statusBar.setBackgroundColor('var(--theia-warn-color0)');
+        this.statusBar.setColor('var(--theia-warn-font-color0)')
```
Comment (kittaakos) [inline packages/core/src/browser/connection-status-service.ts]: Missing semicolon:
```
lerna ERR! [lint] ERROR: /home/travis/build/theia-ide/theia/packages/core/src/browser/connection-status-service.ts[208, 65]: Missing semicolon
```

Comment (lexcorp16) [reply packages/core/src/browser/connection-status-service.ts]: Fixed.

**Comment by kittaakos** [comment]
@lexcorp16, please squash your commits.

**Comment by lexcorp16** [comment]
@kittaakos commits squashed

**Comment by vince-fugnitto** [comment]
@lexcorp16 I've tested your patch but noticed that changing the color like you've done does not actually update the status bar.

selection_067

The intention of the patch was to have the following:

selection_066

This comes from the following CSS selector `#theia-statusBar .area .element`, but your code seems to try and update the parent `#theia-statusBar`

**Comment by kittaakos** [comment]
Thanks for checking @vince-fugnitto, can you please take over the review of this PR?

**Comment by vince-fugnitto** [comment]
@kittaakos yes sure! I have a question however, instead of defining functions as `setColor`, can we not simply define a single `setClassName` which will be either 'online' or 'offline' and have their own custom styling defined in the css?

This will make it a lot easier to customize the status-bar and avoid annoying issues such as hovering over an element when offline.
selection_068

**Comment by kittaakos** [comment]
> yes sure!

👍 

> I have a question however, instead of defining functions as `setColor`, can we not simply define a single `setClassName` which will be either 'online' or 'offline' and have their own custom styling defined in the css?

I do not know if it breaks anything else or not, if you want this change in this PR, you have to figure it out, sorry. I am OK having a `setColor` for now.

**Comment by kittaakos** [comment]
@lexcorp16, please make sure you have signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php) as soon as possible. I have overlooked this in your [PR](https://github.com/theia-ide/theia/pull/3073). If you do not sign it, I have to revert your changes. Ping me here, once you're done. Thank you!

**Comment by lexcorp16** [comment]
@kittaakos Signed.

**Comment by lexcorp16** [comment]
@vince-fugnitto can you please walk me through setting theia up for local testing? So I can verify the impact of my changes

**Comment by vince-fugnitto** [comment]
@lexcorp16 I've made local changes based on the idea I had earlier of only using a function `setClassName` and toggling between `theia-statusBar-online` and `theia-statusBar-offline`. With much less code, I found that it was easier to only use classes and not have to define functions for each css property we want to update. It also meant that hovering was fixed.

status-online

status-offline

**Comment by vince-fugnitto** [comment]
> @vince-fugnitto can you please walk me through setting theia up for local testing? So I can verify the impact of my changes

Sure, all you need to do is start Theia (online status-bar), and close the backend connection (offline status-bar)

**Comment by kittaakos** [comment]
> @kittaakos Signed.

👍 

✅ Verified: _There is a valid ECA on file for afasorojoseph@gmail.com._

**Comment by lexcorp16** [comment]
@vince-fugnitto oh okay. I guess I can close this PR now and you can make one with your changes instead

**Comment by vince-fugnitto** [comment]
@lexcorp16 no its alright, I can step through it with you if you'd like as a learning experience :)

**Comment by lexcorp16** [comment]
@vince-fugnitto absolutely, lets step through it

File: packages/core/src/browser/status-bar/status-bar.tsx [hunk 2]
```diff
@@ -61,15 +61,21 @@ export const STATUSBAR_WIDGET_FACTORY_ID = 'statusBar';
 export const StatusBar = Symbol('StatusBar');
 
 export interface StatusBar {
-    setBackgroundColor(color?: string): Promise<void>;
```
Comment (akosyakov) [inline packages/core/src/browser/status-bar/status-bar.tsx]: please don't break apis

**Comment by vince-fugnitto** [comment]
@lexcorp16 are you still planning to work on this?

**Comment by akosyakov** [comment]
@vince-fugnitto Could you open an alternative PR?

**Comment by vince-fugnitto** [comment]
closing in favour of #3532

**Pull request closed without merging**